### ISSUE TITLE
QuickPick with accessibilitySupport:on on type does not auto select first element if there is no string match

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#93610]
Issue Type: <b>Bug</b>

Linux

1. Launch VScode insiders, have `editor.accessibilitySupport:on`
2. Open a file that contains some lines.
3. Press ctrl + g.
4. Type a number in the range suggested.
5. Press the enter key.

The focus should be moved to the specified line, but in my environment nothing happens.

Isidor comment: The acutal issue is that Quick Pick on type on linux does not select the first element (if there is no substring matching). When enter is hit no element is selected and nothing happens.
Thus I can also reproduce this for Quick Outline for example.

**Comment by bpasero** [comment]
@jvesouza I cannot reproduce this, does it reproduce from a fresh environment when you run `code-insiders --user-data-dir <some empty folder>`?

**Comment by jvesouza** [comment]
@bpasero Yes, the same problem if I use --user-data-dir=jve where jve is an empty directory. However I think I ended up omitting important information in my original report. I am using a screen reader and for this reason the editor.accessibilitySupport needs to be set to on.
The screen reader is the orca since the environment is linux.
Thank you and I'm sorry for not reporting this originally.

**Comment by jvesouza** [comment]
@bpasero I can confirm that the problem happens when editor.accessibilitySupport is  set to on.

I ran code-insiders again using an empty folder and did not activate accessibility.
I pressed ctrl + n to create an empty file and inserted 3 lines in it.
After that I pressed ctrl + g, typed 1 and hit the enter key.
Then I pressed shift + end, ctrl + c, opened a file using gedit and pressed ctrl + v.
The first line inserted in the file created in VSCode was included in the file opened in gedit.

**Comment by isidorn** [comment]
@jvesouza thanks for filling this issue. Yes I can reproduce this, but only on Linux with accessibilitySupport:on. Thus I have changed the original comment and deleted some comments which are not needed. Also assigning to @bpasero since this is caused by the adoption of the new uick pick.

**Comment by bpasero** [comment]
This is a general quick pick issue it seems. I do not have sufficient understanding why we do not select the first item as we do when screen reader is off. I think the solution is to simply accept the first item upon Enter in this case.

Reproduces easily when you configure `"editor.accessibilitySupport": "on"`

**Comment by isidorn** [comment]
Agree. Enter should accept the first element even if not selected. And it would be great if we fix this issue using that approach.
If @chmarti wants to look into the exact cause my suspicion would be that Orca chanes some focus events which the quick pick relies to in this case.

**Comment by joanmarie** [comment]
Will take a look. Thanks!

**Comment by chmarti** [comment]
@isidorn I think you are looking for a different chmarti!

**Comment by jvesouza** [comment]
@isidorn I am not quite sure that the problem is related to the orca. I asked a colleague who uses NVDA to do the same test and according to him, the problem also happened.

**Comment by joanmarie** [comment]
@jvesouza from a quick glance here, there seem to be at least a couple of problems:

1. Pressing Enter after typing the range should cause you to go to that range. But it doesn't. Instead, you must first press Down Arrow in the list and choose the item "Go to line 200." for instance and then press Enter. This issue happens even without Orca running.
2. When you press Down Arrow, and then Enter, VSCode does move you to the specified line (e.g. 200). In addition, pressing Up/Down Arrow after moving to the specified line causes the caret to move amongst the lines/text in the file. But Orca ain't saying a thing. This stands an excellent chance of being an Orca bug.

**Comment by joanmarie** [comment]
@jvesouza Using Orca master, I can no longer reproduce the problem I was seeing and mentioned above. Can you please try the following steps (slightly modified from your opening report):

1. Launch VScode insiders, have editor.accessibilitySupport:on
2. Open a file that contains some lines.
3. Press ctrl + g.
4. Type a number in the range suggested.
5. NEW: Down Arrow once in the listbox to select the one and only item ("Go to line x").
6. Press the enter key.

Expected results: Orca presents the line you just moved to and Up/Down Arrow within that file work as expected.

I get the expected results, whereas I didn't yesterday.

Thanks for checking/testing!

**Comment by jvesouza** [comment]
@joanmarie Confirmed, after press down arrow and enter, orca presents the line X and I can navigate in the file.

**Comment by chrmarti** [comment]
@jvesouza We don't select the first item while in screenreader mode so the placeholder is read out when the picker opens (#57501 reported for NVDA). We could change it to select the first item when filtering, preserving the behavior when opening. (We should also check if NVDA still skips the placeholder when an item is selected, VoiceOver does read it.)

**Comment by bpasero** [comment]
+1 for selecting when filtering, I think at that point the placeholder was read out already.

**Comment by isidorn** [comment]
+1 for selecting when filtering.
Thanks for looking into this!

**Comment by jvesouza** [comment]
It has been fixed. Now, when I type in the line number, orca announces: 'go to line xxx'.
Thanks all.

**Comment by jvesouza** [comment]
I encountered a small problem with the current implementation.
Try the following steps:
1. launch orca, I don't know if the problem happens with a different screen reader.
2. Press f1.
3. Type slowly the following:
lamad.

When I type the d character orca announces the following:
no matching commands, one results.

**Comment by isidorn** [comment]
Awesome. Thanks for letting us know. Adding verified label.
